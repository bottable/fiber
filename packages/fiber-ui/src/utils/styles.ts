import {
  compose,
  space,
  color,
  typography,
  height,
  SpaceProps,
  ColorProps,
  TypographyProps
} from 'styled-system'

export const styleComposition = compose(space, height, color, typography)

export type StyleProps = SpaceProps & ColorProps & TypographyProps
