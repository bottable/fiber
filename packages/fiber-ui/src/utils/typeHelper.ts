export function withProperties<A, B>(component: A, properties: B): A & B {
  Object.keys(properties).forEach((key) => {
    // @ts-ignore
    component[key] = properties[key]
  })
  return component as A & B
}
