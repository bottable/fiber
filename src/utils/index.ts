export * from './typeHelper'
export * from './mergeProps'
