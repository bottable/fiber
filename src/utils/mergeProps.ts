import React from 'react'

export type MergeElementProps<
  T extends React.ElementType,
  P extends object = {}
> = Omit<React.ComponentPropsWithRef<T>, keyof P> & P
