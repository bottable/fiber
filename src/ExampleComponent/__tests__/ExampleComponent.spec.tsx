import { ExampleComponent } from '..'

import React from 'react'
import { render } from 'test/utils'

describe('ExampleComponent', () => {
  it('renders correctly', () => {
    const exampleComponent = render(
      <ExampleComponent
        locale={{ welcome: 'locale overwrite' }}
        mt='516px'
        background='pink'
        text='text'
      />
    )

    expect(exampleComponent).toMatchSnapshot()
  })
})
