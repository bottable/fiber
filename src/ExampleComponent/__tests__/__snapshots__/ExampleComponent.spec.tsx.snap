// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExampleComponent renders correctly 1`] = `
<div
  className="sc-bdnylx jcClSl"
>
  locale overwrite
   
  text
   
  subText
   
</div>
`;
